pkg/datapath/loader/hash_test.go

```diff
@@ -44,6 +46,7 @@ func (s *LoaderTestSuite) TesthashDatapath(c *C) {
 
 	// Ensure we get the same base hash when config is removed via Reset()
 	h.Reset()
+	io.WriteString(h, DatapathSHA256)
```

Comment:
Why is this necessary? Does the test break? The idea behind the "DatapathHash" object is that if you reset it, it goes back to the default SHA so if this test now needs to write the sha inside, then maybe the object isn't resetting correctly (in which case this could indicate a runtime bug).

Reply:
This is necessary because [`newDatapathHash()` seeds the hash with `DatapathSHA256`](https://github.com/cilium/cilium/blob/master/pkg/datapath/loader/hash.go#L43).

I'm a bit suspicious of the datapath hashing code, but I consider it a separate issue to making `go test ./...` to succeed by default, so I've done the minimum changes to get the tests to pass in this PR.

Reply:
:thinking: newDatapathHash() always seeded the hash with DatapathSHA256. 

When I look a little closer, it seems like there is no `h.Reset()` outside of unit tests, so the comment on line 45->47 is actually asserting behaviour that is not implemented and not used. That's part of where this discrepancy came from.  Honestly it looks like we should just remove these 4 lines of test + comment.

I think that maybe at some point, `Reset()` was expected to be implemented and used for resumeable hashes (to ensure that the hash was always seeded with the base datapath sha), but in the end we landed with `sumEndpoint()` which makes a copy of the hash every time, so it's actually not necessary.